protos/events/events.proto

```diff
@@ -0,0 +1,43 @@
+syntax = "proto3";
+
+option go_package = "events";
+
+package github.com.tinkerbell.tink.protos.events;
+
+import "timestamp.proto";
+
+service EventsService {
+  rpc Watch(WatchRequest) returns (stream Event){}
+}
+
+message Event {
+  string id = 1;
+  string resource_id = 2;
+  ResourceType resource_type = 3;
+  EventType event_type = 4;
+  bytes data = 5;
+  google.protobuf.Timestamp created_at = 6;
+}
+
+message WatchRequest {
+  EventType event_type = 1;
+  ResourceType resource_type = 2;
+
+  // Every consumer when it connects to a stream of events will specify
+  // how old the events returned should be (by default 5m).
+  int32 old_events_time_span = 3; // needs a better name
+}
+
+enum ResourceType {
+  TEMPLATE = 0;
+  HARDWARE = 1;
+  WORKFLOW = 2;
+}
+
+enum EventType {
+  CREATED = 0;
+  UPDATED = 1;
+  DELETED = 2;
```

Comment:
These are basic CRUD events which are good, but I think not exhaustive enough to include events for the workflow itself (e.g. IN_PROGRESS, etc). Can we work to include those here?

Reply:
We initially had added events representing changes in workflow state. However, as @mmlb described in his comment [here](https://github.com/tinkerbell/tink/pull/301#discussion_r492278180) the workflow state change should be represented as an update event. The actual state of the workflow will always be present in the data.

Reply:
I think lets keep simple CUD events and we can expand if we want to later, but I doubt we'll want/need to.